src/main/java/org/traccar/protocol/RuptelaProtocolDecoder.java

```diff
@@ -126,9 +135,25 @@ private void decodeParameter(Position position, int id, ByteBuf buf, int length)
                     position.set(Position.KEY_ALARM, Position.ALARM_ACCELERATION);
                 }
                 break;
+            case 170:
+                position.set(Position.KEY_CHARGE, readValue(buf, length, false) > 0);
```

Comment:
This doesn't look right. This should we a charging current. You should always store original values, not do any calculations like this.

Reply:
Ok, I would prefer that too, but some of the KEY_CHARGE decoders do it similarly. Generally speaking KEY_CHARGE is boolean. So either I omit that completely (there is no specific Ruptela IO for charging) or this attribute won't be boolean in this instance (I'm not entirely sure if that is even possible). 
https://github.com/traccar/traccar/blob/7f2ded68ef015684f90bef59c055a1c980688069/src/main/java/org/traccar/protocol/ItsProtocolDecoder.java#L228
https://github.com/traccar/traccar/blob/7f2ded68ef015684f90bef59c055a1c980688069/src/main/java/org/traccar/protocol/JidoProtocolDecoder.java#L118
https://github.com/traccar/traccar/blob/7f2ded68ef015684f90bef59c055a1c980688069/src/main/java/org/traccar/protocol/Gps103ProtocolDecoder.java#L343